File: clients/php/src/Viser/Api/REST/DashboardClientBuilder.php
<?php

namespace Viser\Api\REST;

use Guzzle\Http\Client;

class DashboardClientBuilder
{
    /**
     * @var string
     */
    protected $baseUrl;

    /**
     * @var string
     */
    protected $path;

    /**
     * @var string
     */
    protected $apiVersion;

    public function __construct()
    {
        $this->baseUrl = 'http://localhost:3000';
        $this->path = '/api';
        $this->apiVersion = 'v1';
    }

    /**
     * @param string $baseUrl
     *
     * @return DashboardClientBuilder
     */
    public function setBaseUrl($baseUrl)
    {
        $this->baseUrl = $baseUrl;
        return $this;
    }

    /**
     * @param string $path
     *
     * @return DashboardClientBuilder
     */
    public function setPath($path)
    {
        $this->path = $path;
        return $this;
    }

    /**
     * @param string $apiVersion
     *
     * @return DashboardClientBuilder
     */
    public function setApiVersion($apiVersion)
    {
        $this->apiVersion = $apiVersion;
        return $this;
    }

    /**
     * @return DashboardClient
     */
    public function build()
    {
        $client = new Client($this->baseUrl);

        $dashboardClient = new DashboardClient($client);
        $dashboardClient
            ->setPath($this->path)
            ->setVersion($this->apiVersion);

        return $dashboardClient;
    }
}

File: clients/java/src/main/java/com/github/viser/api/rest/DashboardClientBuilder.java
package com.github.viser.api.rest;

import javax.ws.rs.client.ClientBuilder;
import javax.ws.rs.client.WebTarget;

/**
 * Viser dashboard REST client builder based on JAX-RS client.
 */
public class DashboardClientBuilder
{
    private String baseUrl = "http://localhost:3000";
    private String path = "api";
    private String apiVersion = "v1";

    public void setBaseUrl(String baseUrl) {
        this.baseUrl = baseUrl;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public void setApiVersion(String apiVersion) {
        this.apiVersion = apiVersion;
    }

    /**
     * @return ready to use, configured Viser dashboard client
     */
    public DashboardClient build()
    {
        WebTarget target = ClientBuilder
                .newClient()
                .target(baseUrl)
                .path(path + "/" + apiVersion +"/event");

        return new DashboardClient(target);
    }
}
